### DISCUSSION TITLE
AkashDAO for Decentralized Project Deployment Management

### DISCUSSION THREAD

**Discussion by hyperlander** [akash-network/community#601] [Ideas]
## Introduction
The AkashDAO is a decentralized autonomous organization designed to manage and control the deployment of projects on the Akash Network. This system aims to remove the need for a centralized authority, ensuring that all project deployments and subsequent management actions are governed by smart contracts. By leveraging CosmWasm and Cosmos SDK, AkashDAO will provide a transparent, secure, and automated environment for managing project deployments.

## Objectives
1. **Decentralization**: Eliminate the need for a central authority in project deployments.
2. **Automation**: Use smart contracts to automate the deployment and management process.
3. **Transparency**: Ensure all actions and decisions are recorded on the blockchain, accessible to all DAO members.
4. **Security**: Enhance security through decentralized governance and immutable smart contracts.

## Key Features
1. **DAO Governance**: Utilize a DAO for decision-making processes related to project deployments.
2. **Smart Contract Management**: Deploy and manage projects using smart contracts, ensuring no single entity controls the process.
3. **Voting Mechanism**: Implement a voting system for DAO members to approve or reject deployment proposals.
4. **Resource Allocation**: Automate resource allocation for deployments using predefined smart contract rules.
5. **Auditing and Reporting**: Provide tools for auditing and reporting on deployment activities and resource usage.

## Conclusion
AkashDAO aims to revolutionize the way projects are deployed and managed on the Akash Network by leveraging decentralized governance and smart contracts. This project will ensure that deployments are secure, transparent, and free from centralized control, fostering a more democratic and efficient ecosystem.

**Comment by rodri-r** [comment]
As mentioned in Discord, don't really see what the goal for this is.

- Akash is already decentralized.

- There are no current plans to implement CosmWasm on Akash's network.

- Deployments can be set up to be automated

- The network and everything is already transparent and secured.

- There is already on chain governance.

- There are already auditing and reporting tools available.

**Reply by hyperlander** [reply]
Here is a simple example:

Suppose someone deploys an AI chat service on the Akash Network. This service offers a paid plan. To change the price for this plan, a database update is required. Additionally, updating the UI would necessitate changes to the source code and redeployment.

The idea is to have a DAO manage these pricing and UI updates through smart contracts. In short, the goal is to decentralize the control of specific projects that is deployed on the Akash Network, enabling the distribution of current web2 services.

**Reply by hyperlander** [reply]
In other words, the idea is to control the management of already deployed projects on Akash through DAO (Smart contracts) rather than through the wallets of specific individuals.

**Reply by hyperlander** [reply]
If Akash has no plan forever for this idea, maybe we can get an Akash L2 network that enables Smart contract features.

**Comment by hyperlander** [comment]
Also if Akash had a smart contract feature, it would gain many advantages, such as the ability to transfer service ownership.

**Comment by 31trainman** [comment]
I see potential to use this for something bigger. Send me a dm on akash discord. My discord is 31trainman.

**Reply by Trigs-0** [reply]
I'd be curious to join this conversation as well if we'd like to make it a group chat. I have thoughts to add!

**Comment by Trigs-0** [comment]
Hey @marc-maier I'm currently working with DAODAO on integration potential for an idea like this. As already mentioned the biggest hangup is the lack of CosmWasm on Akash, so the first task at hand is to identify viable use-cases for CosmsWasm on Akash that would justify the state bloat it brings with it. Akash is running lean and mean without it right now, so it will require a significant value-assessment to justify adding simply to enable DAO functionality.

Feel free to keep in touch w/ me on Discord/TG about the idea as I continue to work on it. I'm Trigs_0 on both.
